lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
+      end
+
       def send_request
-        # after method execution the "response" should be in @response_data and should be a string of bytes
+        retries ||= 3
+
+        setup_connection
+        request_options = {:timeout => 60, :ack_timeout => 5}
+        @response_data = nats_request_with_two_responses(cached_subscription_key, @request_data, request_options)
+        parse_response
+      rescue ::NATS::IO::Timeout
+        # Nats response timeout.
+        retry if (retries -= 1) > 0
```

Comment:
should we have a small sleep here? I'm thinking of the case where all the servers are exerting backpressure and what happens if all clients retry a bunch of messages very quickly.